### ISSUE TITLE
Purpur doesn't use spark in cache if download fails on startup.

### ISSUE THREAD

**Issue by Frerduro** [pl3xgaming/Purpur#1324]
### Spark link

unable to right now as this is reporting an issue with spark

### Expected behavior

If download fails use spark stored in cache folder.

### Observed/Actual behavior

When spark download fails it doesn't load spark

### Steps/models to reproduce

Not sure, Maybe block connection to spark download servers temporarily?

### Purpur version

[06:59:02 INFO]: Checking version, please wait...
[06:59:02 INFO]: Current: git-Purpur-1970 (MC: 1.19.4)*
Previous: git-Purpur-1969 (MC: 1.19.4)
* You are running the latest version

### Agreements

- [X] I am running the latest version of Purpur available from https://purpurmc.org/downloads.
- [X] I have searched for and ensured there isn't already an open issue regarding this.
- [ ] I ticked all the boxes without actually reading them
- [X] My version of Minecraft is supported by Purpur.

### Other

[03:35:26] [ServerMain/ERROR]: [SparkProviderSource] Purpur: Failed to download and install spark plugin
[03:35:28] [ServerMain/INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
[03:35:29] [ServerMain/INFO]: Loaded 7 recipes
[03:35:29] [Server thread/INFO]: Starting minecraft server version 1.19.4

![image](https://github.com/PurpurMC/Purpur/assets/35627247/fc8119b7-d41b-416b-b994-33918780784c)

TLDR even if there is a recent download of spark in cache it doesn't load it if something happens when starting your server.

**Comment by encode42** [comment]
Spark will load locally installed versions from the plugins directory. If you're having issues downloading spark, you can just download spark yourself.

I'm not really sure what you're trying to report - does the server *never* use the automatically downloaded version? If the hash of the downloaded version is different than the cached version, it'll attempt to update and downloaded again.

**Comment by Frerduro** [comment]
> Spark will load locally installed versions from the plugins directory. If you're having issues downloading spark, you can just download spark yourself.
> 
> I'm not really sure what you're trying to report - does the server *never* use the automatically downloaded version? If the hash of the downloaded version is different than the cached version, it'll attempt to update and downloaded again.

In trying to say it's not using the cached version. Restarting the server fixed the issue but before it didn't load any spark

**Comment by encode42** [comment]
So the server is trying to download Spark on every start?

**Comment by Frerduro** [comment]
> So the server is trying to download Spark on every start?

yes it is.

**Comment by AdminHcat** [comment]
> So the server is trying to download Spark on every start?

in fact it does, it slows down the startup every time

**Comment by AdminHcat** [comment]
there is no problem after using the symbolic link to put the spark.jar in the plugin directory to cache directory

**Comment by encode42** [comment]
Try again using build `#1974`
